src/app/TargetSelect/CreateTargetModal.tsx

```diff
@@ -47,8 +47,18 @@ export interface CreateTargetModalProps {
 
 export const CreateTargetModal: React.FunctionComponent<CreateTargetModalProps> = (props) => {
   const [connectUrl, setConnectUrl] = React.useState('');
+  const [validConnectUrl, setValidConnectUrl] = React.useState(ValidatedOptions.default);
   const [alias, setAlias] = React.useState('');
 
+  const jmxServiceUrlFormat = new RegExp([
+    /service:jmx:([a-z]+):/
+    ,/\/\/([a-z0-9-]+)\/jndi\/([a-z]+):/
+    ,/\/\/([a-z0-9-]+):([0-9]+)/
+    ,/\/([a-z]+)/g
```

Comment:
When validating JMX service URLs, how specific should the expression matching be? I've allowed any JMX connector and am not sure if we should require `jndi`.

Reply:
It should be fairly lenient. We basically just want to prevent the user from trying to add a connection with an empty URL, or a URL that is obviously not a JMX service URL (like one with http:// scheme, or just some garbled text instead of a URL). Different connectors and transports should not be checked here since the frontend doesn't know what the backend actually supports.

IMO the objective for fixing #348 is just to help prevent users from doing something obviously wrong and possibly confusing, like seeing the "Create" button lit up as if it's clickable (implying their form submission is complete and valid), and then immediately receive an error response because they had forgotten to actually enter a URL.